services/visual-studio-marketplace/visual-studio-marketplace-downloads.service.js

```diff
@@ -42,27 +42,53 @@ module.exports = class VisualStudioMarketplaceDownloads extends VisualStudioMark
         namedParams: { extensionId: 'ritwickdey.LiveServer' },
         staticPreview: this.render({ measure: 'd', count: 1239 }),
         keywords: this.keywords,
+        documentation: `For Azure DevOps extensions, this includes total downloads for both on-prem Azure DevOps Server and Azure DevOps Services`,
       },
     ]
   }
 
-  async handle({ measure, extensionId }) {
-    const json = await this.fetch({ extensionId })
+  transform({ measure, json }) {
     const { statistics } = this.transformStatistics({ json })
     const { value: installs } = this.getStatistic({
       statistics,
       statisticName: 'install',
     })
 
     if (measure === 'i') {
-      return this.constructor.render({ measure, count: installs })
+      return { count: installs }
     }
 
-    const { value: updates } = this.getStatistic({
+    let updates
+    const { value: updateCount } = this.getStatistic({
       statistics,
       statisticName: 'updateCount',
     })
+
+    // updateCount will only be greater than 0 if the extension is for VS or VS Code.
+    // If the value of updateCount is zero then the extension is either:
+    // (A) For VS or VS Code, but has no published updates that have been downloaded
+    // (B) For Azure DevOps
+    // It is not possible to definitively know whether A or B is true, but in either case we should
+    // check the value of the onpremDownloads statistic tracked for Azure DevOps extensions.
+    // If the extension is for VS or VS Code then onpremDownloads will be 0 so we'll still
+    // get the correct number of downloads in all cases.
+    if (updateCount > 0) {
+      updates = updateCount
+    } else {
+      const { value: onpremDownloads } = this.getStatistic({
+        statistics,
+        statisticName: 'onpremDownloads',
+      })
+      updates = onpremDownloads
+    }
```

Comment:
I added the above (lengthy) comment to hopefully clarify the what & why of what's happening here. But basically: although it's not possible to definitively know what type of extension we're dealing with, we _can_ still handle the Azure DevOps extension scenario behind the scenes without requiring the user to specify that explicitly in the route path